lib/opencensus/trace/integrations/sidekiq_middleware.rb

```diff
@@ -0,0 +1,122 @@
+# Copyright 2019 OpenCensus Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "opencensus"
+require "opencensus/trace/integrations/sidekiq"
+require "opencensus/trace/integrations/active_support"
+
+module OpenCensus
+  module Trace
+    module Integrations
+      # # Sidekiq integration
+      #
+      # This is a middleware for Sidekiq applications:
+      #
+      # * It wraps all jobs in a root span
+      # * It exports the captured spans at the end of the job
+      #
+      # ## Configuration
+      #
+      # Example:
+      # # config/initializers/sidekiq.rb
+      #
+      # require "opencensus/trace/integrations/sidekiq_middleware"
+      # Sidekiq.configure_server do |config|
+      #   config.server_middleware do |chain|
+      #     chain.add OpenCensus::Trace::Integrations::SidekiqMiddleware
```

Comment:
We should consider adding a client_middleware since it is the one thing that affects the time of a request to your server.

Reply:
Yes, I could see that being useful to someone. Our Sidekiq jobs are queued by a cron though. Perhaps we should create an issue for it and maybe add it after we have server middleware up and running.